runtime/lua/vim/lsp/util.lua

```diff
@@ -720,18 +720,29 @@ end
 --         ignoreIfExists? bool
 function M.rename(old_fname, new_fname, opts)
   opts = opts or {}
-  local bufnr = vim.fn.bufadd(old_fname)
-  vim.fn.bufload(bufnr)
   local target_exists = vim.loop.fs_stat(new_fname) ~= nil
   if target_exists and not opts.overwrite or opts.ignoreIfExists then
     vim.notify('Rename target already exists. Skipping rename.')
     return
   end
+  local oldbuf = vim.fn.bufadd(old_fname)
+  vim.fn.bufload(oldbuf)
+
+  -- The there may be pending changes in the buffer
+  api.nvim_buf_call(oldbuf, function()
```

Comment:
Feel kinda weird writing buffers, I know it's the right thing to do though... maybe we have an opt that let's you silence an interactive one time prompt before saving? Not sure.

Reply:
I imagine pretty much anybody using this would end up silencing it eventually. 

I'd tend to keep it like this and revisit if somebody ends up complaining.
Keep in mind that it also wrote changes to disk before,  only the approach to it changed a bit